src/averageAges.js

```diff
@@ -15,11 +15,19 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let mans;
+
+  (!century)
+
+    ? mans = people.filter(person => person.sex === 'm')
+
+    : mans = people.filter(person => person.sex === 'm'
+      && Math.ceil(person.died / 100) === century);
+
+  const mansAgeArray = mans.map(man => man.died - man.born);
+
+  return mansAgeArray.reduce((sum, current) =>
+    (sum + current)) / mans.length;
```

Comment:
you do the same on calculateWomenAverageAge. You can create separate function and you it here and on calculateWomenAverageAge